plugins/shared-react/src/utils/pod.ts

```diff
@@ -0,0 +1,6 @@
+import { V1Pod } from '@kubernetes/client-node';
+
+export const shouldStopPolling = (pod: V1Pod) =>
+  pod?.status?.phase === 'Succeeded' ||
+  pod?.status?.phase === 'Failed' ||
+  pod?.status?.phase === 'Running';
```

Comment:
I think we should not stop polling the data when the pipelinerun is in running state so we can see the logs streaming.

```suggestion
  pod?.status?.phase === 'Failed';
```